src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php

```diff
@@ -66,17 +68,63 @@ public function indexAction(WebserviceKeyFilters $filters, Request $request)
 
         $configurationWarnings = $this->lookForWarnings();
 
+        $webserviceConfiguration = $this->get('prestashop.admin.webservice.form_data_provider')->getData();
+        $webserviceIsEnabled = $webserviceConfiguration['webservice_configuration']['enable_webservice'];
+        $webserviceStatus = [
+            'enabled' => $webserviceIsEnabled,
+            'isFunctional' => true,
+            'endpoint' => null,
+        ];
+
+        if ($webserviceIsEnabled) {
+            $expectedWebserviceEntryPoint = $request->getSchemeAndHttpHost() . self::WEBSERVICE_ENTRY_ENDPOINT;
+            $webServiceEntryPointIsReachable = $this->checkWebserviceEndpoint($expectedWebserviceEntryPoint);
+
+            $webserviceStatus['endpoint'] = $expectedWebserviceEntryPoint;
+            $webserviceStatus['isFunctional'] = $webServiceEntryPointIsReachable;
+        }
+
         return $this->render(
             '@PrestaShop/Admin/Configure/AdvancedParameters/Webservice/index.html.twig',
             [
                 'help_link' => $this->generateSidebarLink($request->get('_legacy_controller')),
                 'form' => $form->createView(),
                 'grid' => $presentedGrid,
                 'configurationWarnings' => $configurationWarnings,
+                'webserviceStatus' => $webserviceStatus,
             ]
         );
     }
 
+    /**
+     * @param string $url
+     *
+     * @return bool
+     */
+    private function checkWebserviceEndpoint($url): bool
+    {
+        if ($url === null) {
+            return false;
+        }
+
+        $ch = curl_init($url);
+        curl_setopt($ch, CURLOPT_TIMEOUT, 5);
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 5);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+        $data = curl_exec($ch);
+        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        curl_close($ch);
+
+        if ($httpCode >= 200 && $httpCode < 300) {
+            return true;
+        } elseif ($httpCode == 401) {
```

Comment:
Why 401 is valid?

Reply:
Because the call does not carry an API key. So we expect the API endpoint to return HTTP 401.